pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2553,6 +2561,35 @@ func handlePhysicalIOAdapterListImpl(ctxArg interface{}, key string,
 			// We assume AddOrUpdateIoBundle will preserve any
 			// existing IsPort/IsPCIBack/UsedByUUID
 			aa.AddOrUpdateIoBundle(log, *ib)
+
+			// Adding PF before VF to have correct boot order
+			if ib.Type == types.PhyIoNetEthPF && ib.Vfs.Count > 0 {
+				ifNameMap, err := nettb.GetPciToIfNameMapByTimeout(60 * time.Second)
+				if err != nil {
+					log.Fatal("Failed to obtain ifNameMap")
+				}
+
+				err = createVF(ifNameMap[ib.PciLong], ib.Vfs.Count)
+				if err != nil {
+					log.Fatal("Failed to create VF for iface with PCI address", ib.PciLong)
+				}
+
+				vfs, err := getVfByTimeout(30*time.Second, ifNameMap[ib.PciLong])
```

Comment:
How do we know 30 seconds is sufficient under load?

It seems like we should either wait forever here (which means that the default watchdog for domainmgr will kick in after 300 seconds), or we assume that the VF is set when the pcilong->ifname exists, or we have domainmgr get a notification into the main event loop when the VF information is known.

Wouldn't the VF be set when the pcilong to ifname lookup succeeds? Or does the kernel delay populating the VF info?

Reply:
Creating VFs is almost instant, it PCI bus call. So if it's not done in 30 seconds (which is very generous) something is wrong. But we, of course can wait forever and have big watchdog timer to kill it in case it's not created. Not sure about main event loop notification, since waiting on VFs will delay publishing IoBundle list.

Reply:
I'd rather not do that -- let's be pragmatic.

Reply:
Is it synchronous? If it is asynch and the CPU is 100% busy then how can we know the upper bound?
Waiting forever is fine with me.

Reply:
It is asynchronous